numpy/core/_asarray.py

```diff
@@ -320,10 +357,33 @@ def require(a, dtype=None, requirements=None):
         order = 'C'
         requirements.remove('C')
 
-    arr = array(a, dtype=dtype, order=order, copy=False, subok=subok)
+    arr = array(a, dtype=dtype, order=order, copy=False, subok=subok, like=like)
 
     for prop in requirements:
         if not arr.flags[prop]:
             arr = arr.copy(order)
             break
     return arr
+
+
+# Add documentation for ``like=`` keyword argument
```

Comment:
For these ones I was tempted to let `array_function_dispatch` handle it, but this is fine too.